### PR TITLE
Fix several warnings reported by Microsoft compiler

### PR DESCRIPTION
**Pull request by linev**

### CODE REVIEW

File: core/base/inc/RStipples.h [hunk 1]
```diff
@@ -153,11 +153,16 @@ const unsigned char gStipples[26][32] = {
...
-   }, //25
+   } //25
 };
 
 #else
 
+// Disable warning C4838: conversion from 'int' to 'const unsigned char' requires a narrowing conversion
+
+#pragma warning( push )
+#pragma warning( disable : 4838 )
```
Comment (amadio) [inline core/base/inc/RStipples.h]: This doesn't create problems with other compilers?

File: core/dictgen/res/rootcling_impl.h [hunk 2]
```diff
@@ -41,11 +41,13 @@ namespace RootCling {
 } // namespace Internal
 } // namespace ROOT
 
+#ifndef R__DLLEXPORT
```
Comment (amadio) [inline core/dictgen/res/rootcling_impl.h]: Where else is `R__DLLEXPORT` defined? Shouldn't we be including that header instead?

File: builtins/zlib/CMakeLists.txt [hunk 3]
```diff
@@ -113,8 +113,10 @@ target_include_directories(ZLIB INTERFACE $<BUILD_INTERFACE:${ZLIB_INCLUDE_DIR}>
 if((CMAKE_SYSTEM_PROCESSOR MATCHES "amd64|x86_64|AMD64|X86_64") AND (CMAKE_SYSTEM_NAME MATCHES "Linux"))
    target_compile_options(ZLIB PRIVATE -Wno-unused-function -O3 -mpclmul -Wno-attribute-alias ${ROOT_DEFINITIONS})
 else()
+if(NOT MSVC)
     target_compile_options(ZLIB PRIVATE -O3)
 endif()
+endif()
```
Comment (amadio) [inline builtins/zlib/CMakeLists.txt]: Please use proper indentation.

Comment (linev) [reply builtins/zlib/CMakeLists.txt]: Done

File: core/dictgen/res/rootcling_impl.h (hunk 2 above)
Comment (linev) [reply core/dictgen/res/rootcling_impl.h]: > Where else is R__DLLEXPORTdefined

There is no special include for that. 
Just in approx 5 places same kind of definitions is provided.

File: core/base/inc/RStipples.h (hunk 1 above)
Comment (linev) [reply core/base/inc/RStipples.h]: > This doesn't create problems with other compilers?

It is explicitly for windows - just check few lines before

**Comment by linev** [comment]
I merging this PR and will create same for 6.18 branch

**Comment by linev** [comment]
> The only thing is that stressGraphics has been changed.
> Does it run properly after this change ?

Yes it still works. All changes in stressGraphics and stressHistograms
were about mixing in logical operators bool and int types.

**Comment by couet** [comment]
Ok fine :-)

**Pull request merged**